winpr/libwinpr/utils/wlog/Layout.c

```diff
@@ -123,166 +129,162 @@ static int opt_compare_fn(const void* a, const void* b)
-			                        args[4]);
-			break;
-
-		case 6:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5]);
-			break;
-
-		case 7:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5], args[6]);
-			break;
-
-		case 8:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5], args[6], args[7]);
-			break;
-
-		case 9:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5], args[6], args[7], args[8]);
-			break;
-
-		case 10:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5], args[6], args[7], args[8], args[9]);
-			break;
-
-		case 11:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5], args[6], args[7], args[8], args[9], args[10]);
-			break;
-
-		case 12:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5], args[6], args[7], args[8], args[9], args[10],
-			                        args[11]);
-			break;
-
-		case 13:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5], args[6], args[7], args[8], args[9], args[10],
-			                        args[11], args[12]);
-			break;
-
-		case 14:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5], args[6], args[7], args[8], args[9], args[10],
-			                        args[11], args[12], args[13]);
-			break;
-
-		case 15:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5], args[6], args[7], args[8], args[9], args[10],
-			                        args[11], args[12], args[13], args[14]);
-			break;
-
-		case 16:
-			WLog_PrintMessagePrefix(log, message, format, args[0], args[1], args[2], args[3],
-			                        args[4], args[5], args[6], args[7], args[8], args[9], args[10],
-			                        args[11], args[12], args[13], args[14], args[15]);
-			break;
-	}
+BOOL WLog_Layout_GetMessagePrefix(wLog* log, wLogLayout* layout, wLogMessage* message)
+{
+	char format[WLOG_MAX_PREFIX_SIZE] = { 0 };
+
+	WINPR_ASSERT(layout);
+	WINPR_ASSERT(message);
+
+	char tid[32] = { 0 };
+	SYSTEMTIME localTime = { 0 };
+	GetLocalTime(&localTime);
+
+	struct format_option_recurse recurse = {
+		.options = NULL, .nroptions = 0, .log = log, .layout = layout, .message = message
+	};
+
+#define ENTRY(x) x, sizeof(x) - 1
+	const struct format_option options[] = {
```

Comment:
may I suggest to put the array out of the function, this way we have 1 ident less and possibly most of the entries on one line even after the `clang-format` pass ?

Reply:
no as the struct contains a by call context